spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SourceExample.java

```diff
@@ -0,0 +1,46 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.example;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * @author João André Martins
+ */
+@EnableBinding(Source.class)
+@RestController
+public class SourceExample {
+
+	@Autowired
+	private Source source;
+
+	@PostMapping("/newMessage")
+	public Message sendMessage(@RequestParam("messageBody") String messageBody,
```

Comment:
So, why `@Publisher(Source.OUTPUT)` doesn't work here for you?

Reply:
Humm `@Publisher(Source.OUTPUT)` causes this method to return a 404. Do you have an idea why?

Reply:
Well, no idea, indeed.
Would you mind to share stack trace from logs on the matter?
Thanks

Reply:
Right, there is a proxying.
So, let's never mind for now!
It's not important for this sample.